src/contracts/middleware/BLSOperatorStateRetriever.sol

```diff
@@ -0,0 +1,123 @@
+        uint32[] totalStakeIndices;  
+        uint32[][] nonSignerStakeIndices; // nonSignerStakeIndices[quorumNumberIndex][nonSignerIndex]
+    }
+
+    IBLSRegistryCoordinatorWithIndices public registryCoordinator;
+    IStakeRegistry public stakeRegistry;
+    IBLSPubkeyRegistry public blsPubkeyRegistry;
+    IIndexRegistry public indexRegistry;
+
+    constructor(IBLSRegistryCoordinatorWithIndices _registryCoordinator) {
+        registryCoordinator = _registryCoordinator;
+
+        stakeRegistry = _registryCoordinator.stakeRegistry();
+        blsPubkeyRegistry = _registryCoordinator.blsPubkeyRegistry();
+        indexRegistry = _registryCoordinator.indexRegistry();
+    }
+
+    /**
+     * @notice returns the ordered list of operators (id and stake) for each quorum
+     * @param operatorId the id of the operator calling the function
+     * @return 2d array of operators. For each quorum the provided operaor is a part of, a ordered list of operators
+     */
+    function getOperatorState(bytes32 operatorId) external view returns (Operator[][] memory) {
+        bytes memory quorumNumbers = BitmapUtils.bitmapToBytesArray(registryCoordinator.getCurrentQuorumBitmapByOperatorId(operatorId));
+
+        return getOperatorState(quorumNumbers);
+    }
+
+    /**
+     * @notice returns the ordered list of operators (id and stake) for each quorum
+     * @param quorumNumbers are the ids of the quorums to get the operator state for
+     * @return 2d array of operators. For each quorum, a ordered list of operators
+     */
+    function getOperatorState(bytes memory quorumNumbers) public view returns(Operator[][] memory) {
+        Operator[][] memory operators = new Operator[][](quorumNumbers.length);
+        for (uint256 i = 0; i < quorumNumbers.length; i++) {
+            uint8 quorumNumber = uint8(quorumNumbers[i]);
+            bytes32[] memory operatorIds = indexRegistry.getOperatorListForQuorum(quorumNumber);
+            operators[i] = new Operator[](operatorIds.length);
+            for (uint256 j = 0; j < operatorIds.length; j++) {
+                bytes32 operatorId = bytes32(operatorIds[j]);
+                operators[i][j] = Operator({
+                    operatorId: operatorId,
+                    stake: stakeRegistry.getMostRecentStakeUpdateByOperatorId(operatorId, quorumNumber).stake
+                });
+            }
+        }
+            
+        return operators;
+    }
+
+    /**
+     * @notice returns 
+     *          1) the indices of the quorumBitmaps for each of given operators at the given blocknumber
+     *          2) the indices of the total stakes entries for the given quorums at the given blocknumber
+     *          3) the indices of the stakes of each of the nonsigners in each of the quorums they are a 
+     *             part of (for each nonsigner, an array of length the number of quorums they a part of 
+     *             that are also part of the provided quorumNumbers) at the given blocknumber
```

Comment:
it looks like this is not using _not_ the operators' current sets of quorums (as implied by the wording `quorums they a part of`), but rather the operators' past quorum-sets (i.e. "quorums that they were a part of at the `referenceBlockNumber`")

Reply:
https://github.com/Layr-Labs/eigenlayer-contracts/pull/81/commits/85137533052f6db3db219880c3d77a7f1098fb58
https://github.com/Layr-Labs/eigenlayer-contracts/pull/81/commits/10877ff5ae1d916fee00d413f63a3ac1900b3439